### PR TITLE
[MM-55051] Assign correct timestamp to support packet files

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
See JIRA

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-55051

#### Release Note
```release-note
Assign correct timestamp to support packet files
```

### CODE REVIEW

File: server/channels/app/support_packet.go [hunk 1]
```diff
@@ -94,7 +94,7 @@ func (a *App) generateSupportPacketYaml(c *request.Context) (*model.FileData, er
 	/* LDAP */
 
 	var vendorName, vendorVersion string
-	if ldapInterface := a.ch.Ldap; a.ch.Ldap != nil {
+	if ldapInterface := a.Ldap(); ldapInterface != nil {
```
Comment (hanzei) [inline server/channels/app/support_packet.go:97]: Fix direct access to struct while I'm here

File: server/channels/app/file_test.go [hunk 2]
```diff
@@ -282,18 +285,53 @@ func TestCreateZipFileAndAddFiles(t *testing.T) {
...
 
-	mockBackend = filesStoreMocks.FileBackend{}
-	mockBackend.On("WriteFile", mock.Anything, "directory-to-heaven/zip-file-name-to-heaven.zip").Return(int64(666), nil)
-	err = th.App.CreateZipFileAndAddFiles(&mockBackend, []model.FileData{}, "zip-file-name-to-heaven.zip", "directory-to-heaven")
-	require.NoError(t, err)
+		require.Error(t, err)
+		require.Equal(t, err.Error(), "only those who dare to fail greatly can ever achieve greatly")
+	})
+
+	t.Run("write no file", func(t *testing.T) {
```
Comment (pvev) [inline server/channels/app/file_test.go:303]: question: (haven't gone any deeper), but shouldn't this fail if no files are written? also, if it is working as expected like no failing even when no files are written maybe we can also validate that the file count is zero in the zip file?

**Review by pvev — APPROVED** [review]
LGTM! thank you @hanzei 🚀

File: server/channels/app/file_test.go (hunk 2 above)
Comment (hanzei) [reply server/channels/app/file_test.go:303]: I was also surprised that the code allows for empty archives to be created. While it seems strange to allow that case, I'm 0/5 if it's worth changing the code. We might need that case in the future.

**Review by jespino — CHANGES_REQUESTED** [review]
I think you are changing here the behavior of the zip file generation.

File: server/channels/app/file.go [hunk 3]
```diff
@@ -1383,7 +1383,11 @@ func (a *App) CreateZipFileAndAddFiles(fileBackend filestore.FileBackend, fileDa
 func populateZipfile(w *zip.Writer, fileDatas []model.FileData) error {
 	defer w.Close()
 	for _, fd := range fileDatas {
-		f, err := w.Create(fd.Filename)
+		f, err := w.CreateHeader(&zip.FileHeader{
```
[the remark this answers is not in the document]
Comment (hanzei) [reply server/channels/app/file.go:1386]: Good catch, fixed in a2a377e591

**Review by jespino — APPROVED** [review]
LGTM

**Comment by hanzei** [comment]
@DHaussermann The PR is ready for your review. QA steps are in the ticket.

**Comment by hanzei** [comment]
@yasserfaraazkhan Would you mind taking a look?

**Review by yasserfaraazkhan — APPROVED** [review]
tested this locally and observed date is not 30 November 1979 
Screenshot 2023-11-03 at 1 06 33 PM

**Pull request merged**